lib/vanagon/project.rb

```diff
@@ -82,6 +82,39 @@ def get_files
       files.flatten.uniq
     end
 
+    # Returns a filtered out set of components only including those
+    # components necessary to build a specific component. This is a
+    # recursive function that will call itself until it gets to a
+    # component with no build requirements
+    #
+    # @param name [String] name of component to add. must be present in configdir/components and named $name.rb currently
+    # @return [Array] array of Vanagon::Component including only those required to build "name"
+    def filter_component(name)
+      filtered_component = get_component(name)
+      return nil if filtered_component.nil?
+      included_components = [filtered_component]
+
+      unless filtered_component.build_requires.nil?
```

Comment:
Will `filtered_component.build_requires` be an iterable object like an array even if there's nothing to build? Or will it be another neither-fish-nor-fowl method that returns `nil` **or** an array?